src/game/SpellEffects.cpp

```diff
@@ -446,6 +446,27 @@ void Spell::EffectDummy(SpellEffectIndex eff_idx)
                     m_caster->CastSpell(m_caster, spell_id, true, nullptr);
                     return;
                 }
+                case 8344:                                  // Gnomish Universal Remote (ItemID: 7506)
+                {
+                    Unit* channelTarget = m_originalCaster->GetMap()->GetUnit(m_originalCaster->GetChannelObjectGuid());
```

Comment:
You're absolutely right. If you look at Death ray I have remembered it there, this was a last minute change and I forgot to add in the check after changing it from casting on unitTarget to cast on channelTarget.